src/components/IOUConfirmationList.js

```diff
@@ -328,84 +321,174 @@ class IOUConfirmationList extends Component {
     * @param {Object} option
     */
     toggleOption(option) {
-        // Return early if selected option is currently logged in user.
+        // Return early if selected option is currently logged-in user.
         if (option.login === this.props.session.email) {
             return;
         }
 
-        this.setState((prevState) => {
-            const newParticipants = _.map(prevState.participants, (participant) => {
-                if (participant.login === option.login) {
-                    return {...option, selected: !option.selected};
-                }
-                return participant;
-            });
-            return {participants: newParticipants};
-        });
+        this.setState(
+            (prevState) => {
+                const newParticipants = _.map(prevState.participants, (participant) => {
+                    if (participant.login === option.login) {
+                        return {...option, selected: !option.selected};
+                    }
+                    return participant;
+                });
+                const newSelectedParticipants = _.where(newParticipants, {selected: true});
+                const newSections = this.getSections(newParticipants);
+
+                return {
+                    sections: newSections,
+                    participants: newParticipants,
+                    selectedParticipants: newSelectedParticipants,
+                };
+            },
+            () => { this.allOptions = OptionsListUtils.flattenSections(this.state.sections); },
+        );
+    }
+
+    /**
+     * Scrolls to the focused index within the SectionList
+     *
+     * @param {Number} index
+     */
+    scrollToIndex(index) {
+        if (!this.props.canModifyParticipants) {
+            return;
+        }
+
+        const option = this.allOptions[index];
+        if (!this.list || !option) {
+            return;
+        }
+
+        const {index: itemIndex, sectionIndex} = option;
```

Comment:
**thought (non blocking):** I see we're using a lot of destructuring reassignment here. I don't think it's in our [style guide section about this](https://github.com/Expensify/App/blob/main/STYLE.md#destructuring), but I feel like the reassignment trick is maybe something others have expressed confusion about in the passed so maybe it falls under "reduces code clarity"

Reply:
I don't think there's quite so much of these destructuring alias thingies anymore.